### PR TITLE
Servlet61 Clarify HttpServletMapping getMatchValue

### LINKED ISSUE #26489: Servlet 6.1: Clarify the behaviour of HttpServletMapping.getMatchValue().
`https://github.com/jakartaee/servlet/issues/316`

### PR DESCRIPTION
**Pull request by pmd1nh**
fixes https://github.com/OpenLiberty/open-liberty/issues/26489

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build        #spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.1.internal_fat,io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.6.0_fat_toleration.lWAS,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_part2,com.ibm.ws.webcontainer_fat_servlet31toleration,com.ibm.ws.webcontainer_fat_servlet31toleration_part2,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0_part2,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/.settings/org.eclipse.jdt.ui.prefs [hunk 1]
```diff
@@ -15,46 +16,136 @@ sp_cleanup.add_missing_nls_tags=false
...
+sp_cleanup.use_string_is_blank=false
 sp_cleanup.use_this_for_non_static_field_access=false
 sp_cleanup.use_this_for_non_static_field_access_only_if_necessary=true
 sp_cleanup.use_this_for_non_static_method_access=false
 sp_cleanup.use_this_for_non_static_method_access_only_if_necessary=true
+sp_cleanup.use_unboxing=false
+sp_cleanup.use_var=false
+sp_cleanup.useless_continue=false
+sp_cleanup.useless_return=false
+sp_cleanup.valueof_rather_than_instantiation=false
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/.settings/org.eclipse.jdt.ui.prefs:151]: Are these new settings that we should push out to our other EE11 projects?

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/fat/src/io/openliberty/webcontainer/servlet61/fat/tests/Servlet61HTTPServletMappingTest.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                String responseText = EntityUtils.toString(response.getEntity());
+                LOG.info("\n" + "Response Text: \n[" + responseText + "]");
+
+                if (stringToSearch != null) {
+                    assertTrue("Expected string not found [" + stringToSearch + "]", responseText.contains(stringToSearch));
+                }
+            }
+        }
+    }
+}
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/fat/src/io/openliberty/webcontainer/servlet61/fat/tests/Servlet61HTTPServletMappingTest.java:160]: @pmd1nh  does any of these tests, test the same function as:

1) https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCGetMappingSlashStarTest.java
2) https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WCGetMappingTest.java

If so I think we can reduce the number of tests here.  Let me know.

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/.settings/org.eclipse.jdt.ui.prefs (hunk 1 above)
Comment (pmd1nh) [reply dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/.settings/org.eclipse.jdt.ui.prefs:151]: No, Eclipse keeps reformatting which I don't like it so diable it for the FAT.

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/fat/src/io/openliberty/webcontainer/servlet61/fat/tests/Servlet61HTTPServletMappingTest.java (hunk 2 above)
Comment (pmd1nh) [reply dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/fat/src/io/openliberty/webcontainer/servlet61/fat/tests/Servlet61HTTPServletMappingTest.java:160]: There are 4 new added examples in Java Doc.  The remaining should have been covered by the FAT 4.0 tests.
Updating to only test these 4.

  URI Path (in quotes)         matchValue      pattern         mappingMatch
 "MyServlet/index.html"     ""              /               DEFAULT
 "MyServlet/foo"            ""              /               DEFAULT
 "/bar/foo.extension"       bar/foo         *.extension     EXTENSION
 "/path/foo/bar"            foo/bar         /path/*         PATH

**Comment by pmd1nh** [comment]
#build 
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.1.internal_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

**Comment by pmd1nh** [comment]
@pnicolucci updated.  Please re-review.  Thanks.

**Comment by pmd1nh** [comment]
rebased
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.1.internal_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

**Comment by pmd1nh** [comment]
@pnicolucci  rebased, please re-review.  thanks!

**Comment by pmd1nh** [comment]
#libby

File: dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/ws/webcontainer40/srt/SRTServletRequest40.java [hunk 3]
```diff
@@ -194,8 +194,13 @@ public HttpServletMapping getCurrentHttpServletMapping(WebAppDispatcherContext40
                 case PATH:
                     // in the case of /* mapping, skip the pathInfo.substring.
                     if (!dispatchContext.hasSlashStarMapping()) {
-                        // matchValue is the pathInfo after the last "/" and pattern is the servletPath + "/*"
-                        matchValue = pathInfo.substring(pathInfo.lastIndexOf("/") + 1);
+                        if (com.ibm.ws.webcontainer.osgi.WebContainer.getServletContainerSpecLevel() >= com.ibm.ws.webcontainer.osgi.WebContainer.SPEC_LEVEL_61) {
```
Comment (pmd1nh) [inline dev/com.ibm.ws.webcontainer.servlet.4.0/src/com/ibm/ws/webcontainer40/srt/SRTServletRequest40.java:197]: Doc this behavior change in TODO issue

https://github.com/OpenLiberty/open-liberty/issues/27383

**Review by pnicolucci — APPROVED** [review]

**Pull request merged**